Automattic/woocommerce-payments
# Update XPF currency formatting

Fixes #5150.

#### Changes proposed in this Pull Request

I shared all my findings and decisions on _why_ I'm doing these changes in https://github.com/woocommerce/woocommerce/pull/46960. So here I'll stick to explaining the list of issues that have been flagged and what I did (or didn't) to fix them.

**Issue 1**

>[In the] Currency management screen, the currency preview is producing incorrect previews given my configurations for rounding rules and price charming.

This is no longer an issue, and must have been fixed eventually between the time it's been reported and now. At the moment, the conversion is resulting in the correct price and that can be confirmed by comparing it in any online currency conversion tool.

**Issue 2**

>[In the] Currency management screen, for the Pacific Franc (XPF) the currency symbol is incorrectly placed, and the wrong symbol.

The symbol is being replaced in WC core. As for the formatting, it's being fixed here, and in WC core, through the changes in the currency and locale infos. These changes should be consistent here and in WC core.

The fix for the formatting here is going beyond updating the currency and locale infos. I explain more about it in the comments below.

**Issue 3**

>On the front end of the store, [...] for the XPF currency, they're missing the coma separator for thousands, and also using the incorrect symbol (Trailing Fr, rather than the correct XPF preceding with a space).

The formatting is being fixed here, and in WC core, through the changes in the currency and locale infos. These changes should be consistent here and in WC core.









#### Testing instructions

**Test: XPF currency is formatted correctly in the currency preview**

> [!IMPORTANT]
> This test replaces the test of same name in https://github.com/woocommerce/woocommerce/pull/46960 as we're applying changes in this PR that are modifying the expected outcome. The difference in the outcome is only relevant from WooPayments' perspective, not WooCommerce's.

1. Go through the testing instructions' pre-requisites in https://github.com/woocommerce/woocommerce/pull/46960.
2. Checkout to this branch and build assets: `npm run build:client`.
3. As a merchant, navigate to `WooCommerce > Settings > Multi-currency`.
4. Add the `XPF` currency, if needed.
5. Click on the **manage** link for the XPF currency.
6. In the **Preview** section, the **CFP franc** value should be formatted like `XPF 1,234`.

**Test: XPF currency conversion is working as expected**

1. As a merchant, navigate to `WooCommerce > Settings > Multi-currency`.
2. Add the `XPF` currency, if needed.
3. Click on the **manage** link for the XPF currency.
4. Select `1` for **Price rounding** and note the **CFP franc** value.
5. Go to any online currency conversion tool and perform the same conversion.
7. The price you've got in the conversion tool (step 5) should be close to the price you've got in the currency preview (step 4).

**Test: Another currency (BRL) is formatted correctly in the currency preview**

1. As a merchant, navigate to `WooCommerce > Settings > Multi-currency`.
2. Add the `BRL` currency, if needed.
3. Click on the **manage** link for the BRL currency.
4. In the **Preview** section, the **Brazilian real** value should be formatted like `R$ 1.234,56`. Without these changes, it was previously being formatted like `R$1,234.56`, and from the Brazilian perspective, that is not correct.







-------------------


**Post merge**
